doc/stdenv/meta.chapter.md

```diff
@@ -134,7 +134,27 @@ Attribute Set `lib.platforms` defines [various common lists](https://github.com/
 This attribute is special in that it is not actually under the `meta` attribute set but rather under the `passthru` attribute set. This is due to how `meta` attributes work, and the fact that they are supposed to contain only metadata, not derivations.
 :::
 
-An attribute set with as values tests. A test is a derivation, which builds successfully when the test passes, and fails to build otherwise. A derivation that is a test needs to have `meta.timeout` defined.
+An attribute set with tests as values. A test is a derivation that builds when the test passes and fails to build otherwise.
+
+#### Regular tests
+
+Tests that are part of the source package are often executed in the `installCheckPhase`. Prefer `passthru.tests` for tests that are introduced in nixpkgs because `installCheckPhase` adds overhead to each build and we can run `passthru.tests` independently.
+
+```nix
+{ /* ... */, key }:
+
+mkDerivation rec {
+  # ...
+  passthru.tests = {
+    version-check = runCommand "key-help" {} ''
+      ${key}/bin/KeY --help | grep -Fw ${version}
+      touch $out
+    '';
+  };
+}
+```
+
```

Comment:
I [mentioned this out in the main thread](https://github.com/NixOS/nixpkgs/pull/119731#issuecomment-822089022), but I guess I'll attach it here since you pinged me :)

> Without a resource covering what to test, I imagine some readers will see the help/version example here as a straightforward example of a test that is good/desirable or at least acceptable. It might be good to drop some breadcrumbs about where it sits on the various scales (Is it just a trivial example that was once acceptable but would get laughed out of PR review these days? Is it the minimum-meaningful test, or all we should aspire until everything is at this level?)
> 
> If it is trivial, saying so is probably enough. If it is meaningful, a quick high-level note on ~why it's meaningful might be a good stop-gap for orienting people towards appropriate values?

I think it would be good to have a little context on this before or after the example, but I don't see it as a blocker and realize you might not know the answer.

Reply:
> but I don't see it as a blocker and realize you might not know the answer.

Correct, I don't know if we have enough experience/consensus to be able to put into words authoritative advice on this yet, so I left it out for now.